src/js/controllers/mapController.ts

```diff
@@ -156,23 +156,59 @@ export class MapController {
 
           const mapLayerObjects: LayerProps[] = [];
           this._map?.layers.forEach((layer: any) => {
```

Comment:
@vaidotasp what are your thoughts on moving the `.forEach()` into a designated helper function? I noticed the logic inside of this `.forEach()` is used on lines 158 and 435. Could be helpful if we need to update the logic later on?

Reply:
good idea. check the commit 8125afef323378a573f9ed78283991393920ed09